### ISSUE TITLE
Fails to retrieve a page from esolangs.org -- gets "The action specified by the URL is invalid" for Query {"meta"=>"siteinfo"}

### ISSUE THREAD

**Issue by Nakilon** [molybdenum-99/infoboxer#84]
```
Infoboxer.wiki('https://esolangs.org/').get('Category:Implemented')
=> #<JSON::ParserError: 784: unexpected token at '<!DOCTYPE html>
```

The page says:

> The action specified by the URL is invalid.

Byebug says the request is `#<MediaWiktory::Wikipedia::Actions::Query {"meta"=>"siteinfo"}>`

**Comment by zverok** [comment]
```ruby
Infoboxer.wiki('https://esolangs.org/w/api.php').get('Category:Implemented')
# => #<Page(title: "Category:Implemented", url: "https://esolangs.org/wiki/Category:Implemented"): The languages in this category ...>
```
The URL should be the base URL for the API. It can be `/w/api.php`, or just `/api.php`, or sometimes `/wiki/api.php`, so we don't try to guess. Though the error message should definitely be more obvious! (Like, check the most basic response and say "It is not an API endpoint" if it is wrong)

**Comment by Nakilon** [comment]
Oh, the `/w/api.php` wasn't obvious because README says:
```
# From any wiki with the same engine:
page = Infoboxer.wiki('http://companywiki.com').get('Our Product')
```
After creating this ticket I used another gem like this:
```
MediaWiki::Butt.new "https://esolangs.org/w/api.php"
```
that needs this suffix too but I didn't realise I could come back to the Infoboxer gem with this edit because I didn't even notice it since I'm not generally familiar with Mediawiki API.

**Comment by zverok** [comment]
Yeah, that was clear once (in docs and errors), and then on one of the last big redesign, it got lost :( Sorry.

I'll keep it as a reminder, yeah.
